siatest/renter/renter_test.go

```diff
@@ -75,13 +76,23 @@ func testUploadDownload(t *testing.T, tg *siatest.TestGroup) {
 		t.Fatal(err)
 	}
 	// Download the file asynchronously and wait for the download to finish.
-	localFile, err := renter.DownloadToDisk(remoteFile, true)
+	localFile, err = renter.DownloadToDisk(remoteFile, true)
 	if err != nil {
 		t.Error(err)
 	}
 	if err := renter.WaitForDownload(localFile, remoteFile); err != nil {
 		t.Error(err)
 	}
+	// Stream the file.
+	_, err = renter.Stream(remoteFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Stream the file partially.
+	_, err = renter.StreamPartial(remoteFile, localFile, 0, uint64(fileSize/2))
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Would like to see a loop,  5 calls to StreamPartial, from a random starting point to a random ending point that is somewhere between the starting point and end of the file.

Otherwise this PR is looking pretty good.